### ISSUE TITLE
Error on bytecoded functions

### ISSUE THREAD

**Issue by trahelyk** [SESYNC-ci/rslurm#14]
This might be down to my limited understanding of both bytecoding and slurm, but I've noticed that the slurm engine throws an error (specifically, “Error: unexpected '<' in "<" Execution halted”) any time the parallelized function is bytecoded. This is because any bytecoded function returns from capture.output() with a <bytecode> tag in its tail. It isn't clear to me how to prevent a function of a certain complexity from becoming byte coded (perhaps an environment setting?). The function I’m trying to parallelze contains a call to lm(), which seems to be what is causing it to bytecode. At any rate, I can prevent this problem by creating a utility that strips the “<bytecode>” text from the capture.output() results:
 
```
strip_bytecode <- function(func)
{
  func_text <- capture.output(func)
  if(any(grepl("bytecode:", tail(capture.output(func),2)))) {
    x <- func_text[1:(which(grepl("bytecode:", func_text))-1)]
  } else {
    x <- func_text
  }
  return(x)
}
``` 
And then modifying the func_to_str() function to call that utility:
 
```
func_to_str2 <- function(f) {
  fstr <- paste(strip_bytecode(f), collapse = "\n")
  gsub("<environment: [A-Za-z0-9]+>", "", fstr)
}
```

Should I be doing something else to prevent the bytecode?

**Comment by itcarroll** [comment]
Though I was unaware of this issue, I believe it is already resolved in the next rslurm release. We had switched from passing functions as strings to passing functions as objects to Slurm for other reasons, but a quick test (see 5cabe18) suggests this resolves the bytecoded function problem too.

Would you test whether it resolves the issue you describe? Use the `devtools` package to install the latest version of rslurm (with `devtools::install_github('sesync-ci/rslurm')`), and report your findings here. Thanks very much for reporting!

**Comment by itcarroll** [comment]
Oops, did not mean to close yet!

**Comment by trahelyk** [comment]
Yes, this resolves the issue in my code as well. Thank you!

**Comment by itcarroll** [comment]
Thank you for testing. The fix will be published in version 0.4.0 on CRAN, any day now.
